app/src/main/java/ch/epfl/sweng/swenggolf/database/FireDatabase.java

```diff
@@ -205,18 +205,47 @@ public void readOffers(@NonNull final ValueListener<List<Offer>> listener,
             listener.onDataChange(new ArrayList<Offer>());
         }
         for (int i = 0; i < categories.size(); ++i) {
+            // TODO change the ordering to put the most recent on top?
             Query query = ref.orderByChild("tag").equalTo(categories.get(i).toString());
             readListQuery(listener, query, Offer.class, false);
         }
     }
 
+    @Override
+    public void readFollowers(@NonNull final ValueListener<Map<String, List<String>>> listener) {
+        checkInternetConnectionValue(listener);
+        final DatabaseReference ref = database.getReference(FOLLOWERS_PATH);
+        ref.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                Map<String, List<String>> usersFollowing = new HashMap<>();
```

Comment:
I don't know if using a TreeMap here would be more efficient, what do you think ?

Reply:
I don't see the advantages to have the keys sorted, but if you think it's better I can change it.

Reply:
Maybe it's quicker for getting the elements, and since the list is pretty huge it might be efficient ?

Reply:
Since you're only iterating once through them let's keep an HashMap.